client/src/pages/Report/OnDemandReport.tsx

```diff
@@ -21,19 +22,57 @@ const OnDemandReport = () => {
     const [nodes, setNodes] = useState<NodeDataRow[]>([]);
     const [accuracy, setAccuracy] = useState<AccuracyLevel>('HIGH');
     const navigate = useNavigate();
+    const [startDateMs, setStartDateMs] = useState<number>(Date.now());
+    const [endDateMs, setEndDateMs] = useState<number>(Date.now());
+    const [showInfoPopup, setShowInfoPopup] = useState(false);
+
+    const handleDateRangeChange = (startMs: number, endMs: number) => {
+        setStartDateMs(startMs);
+        setEndDateMs(endMs);
+    };
+
+    const filterNotificationChannels = (channels: NotificationChannel[]) => {
+        const slackReceiverIds: number[] = [];
+        const discordReceiverIds: number[] = [];
+        const mailReceiverIds: number[] = [];
+
+        channels.forEach((channel) => {
+            const channelId = parseInt(channel.id, 10);
+            if (!isNaN(channelId)) {
+                switch (channel.service.toLowerCase()) {
+                    case 'slack':
+                        slackReceiverIds.push(channelId);
+                        break;
+                    case 'discord':
+                        discordReceiverIds.push(channelId);
+                        break;
+                    case 'email':
+                        mailReceiverIds.push(channelId);
+                        break;
```

Comment:
Aren't service supposed to be somehting like literal type (something of an expected value). Couldn't we use that instead of making it LowerCase and comparing against strings?